engine/src/main/java/io/zeebe/engine/processing/bpmn/task/ServiceTaskProcessor.java

```diff
@@ -53,59 +54,49 @@ public ServiceTaskProcessor(final BpmnBehaviors behaviors) {
   }
 
   @Override
-  public void onActivating(final ExecutableServiceTask element, final BpmnElementContext context) {
-
+  public void onActivate(final ExecutableServiceTask element, final BpmnElementContext context) {
     variableMappingBehavior
         .applyInputMappings(context, element)
         .flatMap(ok -> eventSubscriptionBehavior.subscribeToEvents(element, context))
+        .flatMap(ok -> evaluateJobExpressions(element, context))
         .ifRightOrLeft(
-            ok -> stateTransitionBehavior.transitionToActivated(context),
+            tuple -> {
+              createNewJob(context, element, tuple.getLeft(), tuple.getRight().intValue());
+              stateTransitionBehavior.transitionToActivated(context);
+            },
```

Comment:
It is not clear at this level what tuple's left and right are, while they feel important at this abstraction level. Some ideas I see would be: rename tuple to `jobTypeAndRetries`; introduce named local variables for the left and right values; or introduce an inner class decorating the tuple with `getJobType` and `getRetries` methods. Perhaps you see an even better way. Having said that, the tuple structure it self is definitely an improvement over the previous code.

Reply:
I also rename the tuple variable. But it seems I lost the change on merging :sweat_smile: